src/main/java/org/eclipsefoundation/marketplace/dao/impl/DefaultSolrDao.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (C) 2019 Eclipse Foundation and others.
+ * 
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+*/
+package org.eclipsefoundation.marketplace.dao.impl;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+
+import org.apache.solr.client.solrj.SolrQuery;
+import org.apache.solr.client.solrj.SolrServerException;
+import org.apache.solr.client.solrj.impl.HttpSolrClient;
+import org.apache.solr.client.solrj.impl.XMLResponseParser;
+import org.apache.solr.client.solrj.response.QueryResponse;
+import org.apache.solr.client.solrj.response.UpdateResponse;
+import org.apache.solr.common.SolrDocument;
+import org.apache.solr.common.SolrException;
+import org.apache.solr.common.SolrInputDocument;
+import org.apache.solr.common.util.NamedList;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.eclipsefoundation.marketplace.dao.SolrDao;
+import org.eclipsefoundation.marketplace.dao.mapper.SolrBeanMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Default implementation of the Solr DAO interface. Creates a reusable client
+ * object to maintain sessions and latches to a single Solr core.
+ * 
+ * @author Martin Lowe
+ */
+@ApplicationScoped
+public class DefaultSolrDao implements SolrDao {
+	private static final Logger LOGGER = LoggerFactory.getLogger(DefaultSolrDao.class);
+
+	@ConfigProperty(name = "solr.url.base")
+	String baseUrl;
+
+	@ConfigProperty(name = "solr.core")
+	String core;
+
+	private HttpSolrClient client;
+
+	/**
+	 * Initializes the Solr client object with the injected properties from
+	 * micro-profile. The parser is set manually as by default there is none set.
+	 * This doesn't affect the data returned, only how it is serialized.
+	 */
+	@PostConstruct
+	public void init() {
+		LOGGER.debug("Creating Solr client for {}/{}", baseUrl, core);
+
+		this.client = new HttpSolrClient.Builder(baseUrl + '/' + core).build();
+		this.client.setParser(new XMLResponseParser());
+	}
+
+	@Override
+	public <T, M extends SolrBeanMapper<T>> List<T> get(SolrQuery q, M mapper) {
+		List<T> beans = new LinkedList<>();
```

Comment:
Why do you use LinkedList? Default implementation should be ArrayList, any other choice should be justified (e.g. LinkedList because you want to insert at certain indexes).

Also, this variable is not defined in the proper scope. It should be defined within the try {} block as it's not used outside.

Anyway, this will be removed if you follow my next comment.